packages/build/src/plugins_core/functions/utils.js

```diff
@@ -29,14 +29,11 @@ export const getUserAndInternalFunctions = ({
   frameworkFunctionsSrcExists,
 }) => {
   const paths = [
+    frameworkFunctionsSrcExists ? frameworkFunctionsSrc : undefined,
     functionsSrcExists ? functionsSrc : undefined,
     internalFunctionsSrcExists ? internalFunctionsSrc : undefined,
   ]
```

Comment:
Looks like order matter here given that we restructure array into specific ~categories here - https://github.com/netlify/build/blob/4589f162e6509dc0d17f3e079921b76faf46b7aa/packages/build/src/plugins_core/functions/index.ts#L139

Which I think is what causing this test failure https://github.com/netlify/build/actions/runs/13266394208/job/37034850085?pr=6053#step:12:791

So probably should be

```suggestion
  const paths = [
    functionsSrcExists ? functionsSrc : undefined,
    internalFunctionsSrcExists ? internalFunctionsSrc : undefined,
    frameworkFunctionsSrcExists ? frameworkFunctionsSrc : undefined,
  ]
```

(or alternatively change the signature of the function to return object with named fields/keys and adjust places we call that function)